### DISCUSSION TITLE
How to bypass UTM and run QEMU under the hood on Mac M1?

### DISCUSSION THREAD

**Discussion by jamesmortensen** [utmapp/UTM#3128] [UTM Help (macOS)]
UTM on the Mac M1 has been a lifesaver for virtualization.  I wanted to learn more about QEMU and have attempted to get it running on my Mac M1; however, I've not been successful, even following various instructions regarding applying QEMU patches to add the necessary hypervisor support.  

Since UTM is using QEMU, I assumed I'd find a qemu-system-aarch64 binary buried somewhere in the application package, but I have not found it.  I see Docker has one, and I assume QEMU is how Docker is able to run on the Mac OS.

Can someone shed light on how UTM uses QEMU?  I'd like to be able to start my VMs from the terminal to learn more about QEMU and am assuming that there is a binary packaged with UTM that already has all of the necessary patches applied for the Mac M1.  

Here is what I find on my system:

```
$ locate qemu-system-aarch64
/Applications/Docker.app/Contents/MacOS/qemu-system-aarch64
~/Dev/ACVM/ACVM/qemu-system-aarch64
~/Dev/virtualization/libvirt-qemu/qemu-ubuntu/qemu/build/aarch64-softmmu/qemu-system-aarch64
/opt/homebrew/Cellar/qemu/6.0.0/bin/qemu-system-aarch64
/opt/homebrew/bin/qemu-system-aarch64
```

ACVM comes with QEMU, and the one in the libvirt-qemu folder is from my failed attempt to patch QEMU, and the one installed in brew I don't think is being used by UTM at all, especially since I see v6.1 in the UTM interface, and the homebrew qemu-system-aarch64 binary is in v6.0.0 folder.

**Comment by saagarjha** [comment]
UTM builds QEMU into a dynamic library. You can find these inside the Frameworks/ folder of the bundle.

**Reply by jamesmortensen** [reply]
Thanks for the reply.  I previously poked around in the Frameworks folder, but the qemu-system-aarch64 binary is not there.

**Comment by osy** [comment]
If you download the sysroot from GitHub you can run the ./bin/qemu* binaries. But if it complains about dylibs not found you may need to symlink them to /Users/runner/UTM

**Reply by jamesmortensen** [reply]
Thanks for the reply. By "download sysroot", do you mean the artifact that's produced in the UTM build process in GitHub Actions?  If so, is it okay if I get the one from the latest build or does it have to be from the 2.3.4 (latest at the time of this post)?

<img width="961" alt="Screenshot 2021-09-28 at 3 42 34 PM" src="https://user-images.githubusercontent.com/1315816/135068364-eecd5ffa-0ea6-47b7-9e53-a09cfeef0ec0.png">
